### ISSUE TITLE
`error: collection is never read` on `#[derive(Deserialize)]`

### ISSUE THREAD

**Issue by epilys** [serde-rs/serde#2789]
On both current stable 1.80.0 and nightly 1.82.0-nightly (2024-08-06 60d146580c10036ce89e)

sample code:

```rust
#![deny(clippy::collection_is_never_read)]

use serde::Deserialize;

#[derive(Deserialize)]
struct Options {
    a: bool,
}

#[derive(Deserialize)]
#[serde(untagged)]
enum Inner<'a> {
    Default {
        s: &'a str,
    },
    Builder {
        s: &'a str,
        #[serde(flatten)]
        o: Options,
    },
}
```

Getting:

```text
error: collection is never read
  --> src/lib.rs:10:10
   |
10 | #[derive(Deserialize)]
   |          ^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collection_is_never_read
note: the lint level is defined here
  --> src/lib.rs:1:9
   |
1  | #![deny(clippy::collection_is_never_read)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
```

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d82d6da1e402e39a9c4b30b72b5a6074)

**Comment by dtolnay** [comment]
Thanks! Fixed in serde_derive 1.0.205.

**Comment by epilys** [comment]
Thank _you_, too, @dtolnay :)
